app/views/charge.njk

```diff
@@ -321,152 +321,161 @@
                   <img src="/images/amex-security-code.png" class="amex-cvc hidden" alt="{{ __p("cardDetails.amexcvcTip") }}"/>
                 </div>
                 {% if collectBillingAddress %}
-                  <div class="govuk-form-group govuk-!-width-three-quarters govuk-!-padding-top-4 govuk-!-margin-top-8 pay-!-border-top {% if highlightErrorFields.addressCountry %} govuk-form-group--error{% endif %}" data-validation="addressCountry">
-                    <legend for="address-country" class="govuk-!-margin-bottom-6">
-                      {% if allowApplePay or allowGooglePay %}
-                        <h3 class="govuk-heading-m govuk-!-margin-bottom-1 web-payment-button-heading">{{ __p("cardDetails.billingAddress") }}</h3>
-                      {% endif %}
-                      <h2 class="govuk-heading-m govuk-!-margin-bottom-1 non-web-payment-button-heading">{{ __p("cardDetails.billingAddress") }}</h2>
+                  <div class="govuk-!-width-three-quarters govuk-!-padding-top-4 govuk-!-margin-top-8 pay-!-border-top">
+                      <fieldset class="govuk-fieldset" aria-describedby="address-hint">
+                        <legend>
+                          {% if allowApplePay or allowGooglePay %}
+                            <h3 class="govuk-heading-m govuk-!-margin-bottom-1 web-payment-button-heading">{{ __p("cardDetails.billingAddress") }}</h3>
+                          {% endif %}
```

Comment:
could you explain to me what the purpose of this bit is (I know this isn't really a change you've made, just curious)

Reply:
The Apple & Google pay code can get confusing.

`{% if allowApplePay or allowGooglePay %}` - This is a flag that states whether the PSP has Google or Apple pay enabled.  This is set in the Node code.

Then the `H3` heading tag is displayed with the text `Billing Address`.
When Apple or Google pay is not present - then this heading is set as a `H2` instead further down the page.

Please let me know if you have any more questions.

Reply:
I was more wondering why

Reply:
The layout of the `Payment Page / Charge page` changes depending on if the `Apple / Google` Pay buttons are visible.

We have an extra **heading**  when the Apple/Google pay buttons are visible.  So all the existing headings will need to drop down a heading level.  I.e. A `h2` will need to become a `h3`.

Please let me know if you have any more questions.